src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -400,6 +401,27 @@ class spell_warr_rallying_cry : public SpellScript
     }
 };
 
+// 385952 - Shield Charge
+class spell_warr_shield_charge : public SpellScript
+{
+    PrepareSpellScript(spell_warr_shield_charge);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_WARRIOR_SHIELD_CHARGE_EFFECT });
+    }
+
+    void HandleDummy(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_WARRIOR_SHIELD_CHARGE_EFFECT);
```

Comment:
Are you sure you don't need any TriggerCastFlags?

Reply:
Yes